presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -448,6 +449,10 @@ private void addToHiveSplitSourceRoundRobin(List<Iterator<HiveSplit>> iteratorLi
 
     private static List<LocatedFileStatus> listAndSortBucketFiles(HiveFileIterator hiveFileIterator, int bucketCount)
     {
+        if (hiveFileIterator.hasPathPredicate()) {
```

Comment:
Per @electrum  suggestion, I've moved the check to the HiveFileIterator. There is side effect though for bucketed tables. The current code in the Hive connector, assumes a list that will contain all the files in the bucket, so you could directly target a single file (thus in a way, already having a targeted file predicate). My initial instinct was to ignore the predicate if it was a bucketed table, but I find that misleading to the user. I opted instead to throw an exception for the case of a bucketed table with the a $path predicate.